File: ffrontend/src/app/commons/dtos/DtoProgrammingAdd.ts
export interface BBProgrammingAdd{
    jobId : number;
    name : string;
    round : number;
    title : string;
    description : string;
    language : string;
    timeLimit : number;
    codeTemplate : string;
    testCases : Array<BTestCase>;
}

export interface BTestCase{
    input : string;
    expectedOutput : string;
}

export interface RSolveP{
    kerdoId : number;
    title : string;
    description : string;
    language : string;
    codeTemplate : string;
    kezdesIdo : Date;
    befejezesIdo : Date;
    kitoltesPerc : number;
}

export interface RKitoltottPTeszteset{
    bemenet : string;
    elvartKimenet : string;
    stderr : string;
    memoriaMB : number;
    futasidoms : number;
    helyes : boolean;
    ellenorizve : boolean;
    nemfutle : boolean;
}

export interface RKitoltottP{
    kerdoivId: number;
    title: string;
    description: string;
    language?: string;
    codetemplate?: string;
    kezdesIdo: Date;
    befejezesIdo: Date;
    kitoltesPerc: number;
    tesztesetek: RKitoltottPTeszteset[];
}

File: allasinterju-backend/Allasinterju.Core/DTOs/DtoProgrammingAdd.cs
public class BProgrammingAdd{
    public int JobId{get;set;}
    public string Name{get;set;}
    public int Round{get;set;}
    public string Title{get;set;}
    public string Description{get;set;}
    public string Language{get;set;}
    public int TimeLimit{get;set;}
    public string CodeTemplate{get;set;}
    public List<BTestCase>? TestCases{get;set;}

}
public class BTestCase{
    public string Input{get;set;}
    public string ExpectedOutput{get;set;}
}
